shared/chat/conversation/info-panel/notifications/container.js

```diff
@@ -1,157 +1,155 @@
+    props => ({
+      channelWide: props._storeChannelWide,
+      desktop: props._storeDesktop,
+      mobile: props._storeMobile,
+      muted: props._storeMuted,
+      saveState: 'same',
+    }),
+    {
+      toggleChannelWide: (state, props) => () => {
+        props._updateNotifications(state.desktop, state.mobile, !state.channelWide)
+        return {channelWide: !state.channelWide}
+      },
+      toggleMuted: (state, props) => () => {
+        props._muteConversation(!state.muted)
+        return {muted: !state.muted}
+      },
+      updateDesktop: (state, props) => (desktop: Types.NotificationsType) => {
+        if (desktop === state.desktop) {
+          return {}
+        }
+        props._updateNotifications(desktop, state.mobile, state.channelWide)
+        return {desktop}
+      },
+      updateMobile: (state, props) => (mobile: Types.NotificationsType) => {
+        if (mobile === state.mobile) {
+          return {}
+        }
+        props._updateNotifications(state.desktop, mobile, state.channelWide)
+        return {mobile}
+      },
+      updateSaveState: () => (saveState: SaveStateType) => ({saveState}),
+      syncLocalToStore: (state, props) => (channelWide, desktop, mobile, muted) => ({
+        channelWide,
+        desktop,
+        mobile,
+        muted,
+      }),
+    }
+  ),
   lifecycle({
-    componentDidMount() {
-      this.props._resetSaveState()
+    componentWillReceiveProps(nextProps) {
+      // Same as the store?
+      if (
+        nextProps._storeDesktop === nextProps.desktop &&
+        nextProps._storeMobile === nextProps.mobile &&
+        nextProps._storeChannelWide === nextProps.channelWide &&
+        nextProps._storeMuted === nextProps.muted
+      ) {
+        // Mark it as saved
+        if (nextProps.saveState === 'saving') {
+          nextProps.updateSaveState('justSaved')
+          setTimeout(() => {
+            nextProps.updateSaveState('same')
+          }, 300)
+        }
+      } else {
+        // did our local settings change at all?
+        if (
+          this.props.desktop !== nextProps.desktop ||
+          this.props.mobile !== nextProps.mobile ||
+          this.props.channelWide !== nextProps.channelWide ||
+          this.props.muted !== nextProps.muted
+        ) {
+          if (nextProps.saveState !== 'saving') {
+            nextProps.updateSaveState('saving')
+          }
+        }
+      }
+      // store changed?
+      if (
+        this.props._storeDesktop !== nextProps._storeDesktop ||
+        this.props._storeMobile !== nextProps._storeMobile ||
+        this.props._storeChannelWide !== nextProps._storeChannelWide ||
+        this.props._storeMuted !== nextProps._storeMuted
+      ) {
+        nextProps.syncLocalToStore(
+          nextProps._storeChannelWide,
+          nextProps._storeDesktop,
+          nextProps._storeMobile,
+          nextProps._storeMuted
+        )
+      }
     },
-  })
+  }),
+  setDisplayName('LifecycleNotifications')
```

Comment:
🍳 This should be right after `connect` if the goal is to make it work with storybook

Reply:
not sure what you mean about this. don't you actually want this to be last?